### PR TITLE
Use system/vcpkg spdlog when available otherwise fallback to fetchcontent

### PR DESCRIPTION
**Pull request by 9prady9**
Description
-----------
This change enables using system/vcpkg spdlog when available otherwise fallback to fetchcontent workflow. I have also udpated the spdlog version to 1.8.5 from the archaic 1.0.0 version. Fortunately, there were no issues like we encountered with fix releases during the version update of spdlog.

Apart from this, I have corrected the extern arrayfire deps download location; This is just a folder organization change, didn't fix any build issue.

Changes to Users
----------------
None

Checklist
---------
- [x] Rebased on latest master
- [x] Code compiles
- [x] Tests pass
- ~[ ] Functions added to unified API~
- ~[ ] Functions documented~

### CODE REVIEW

**Comment by 9prady9** [comment]
Turns out the issue is with the dependency of spdlog, fmt whose latest version on vcpkg has compilation issues. I am working on a workaround. Will update the PR once it is ready

**Comment by 9prady9** [comment]
the work around seemed to work on local windows setup, I wonder why it failed on CI after the change - I will look into the windows problem.

**Comment by 9prady9** [comment]
> 
> 
> the work around seemed to work on local windows setup, I wonder why it failed on CI after the change - I will look into the windows problem.

Never mind, I think this is due to a100 system being blocked by me for running another benchmark. I will shut that down now that I don't need it further.

File: CMakeLists.txt [hunk 1]
```diff
@@ -247,7 +254,11 @@ else()
                              ${ArrayFire_SOURCE_DIR}/include
                              ${ArrayFire_BINARY_DIR}/include
                              ${ArrayFire_SOURCE_DIR}/src/backend)
-  target_link_libraries(bin2cpp PRIVATE spdlog)
+  if(TARGET spdlog::spdlog_header_only)
```
Comment (umar456) [inline CMakeLists.txt:257]: af_spdlog?

Also why is bin2cpp using spdlog anyway. It doesn't look like it uses it.

Comment (9prady9) [reply CMakeLists.txt:257]: bin2cpp uses `common/util.hpp` and that indirectly causes this I think.

**Review by umar456 — APPROVED** [review]

**Pull request merged**